perun-core/src/main/java/cz/metacentrum/perun/core/entry/ResourcesManagerEntry.java

```diff
@@ -99,6 +99,32 @@ public Resource createResource(PerunSession sess, Resource resource, Vo vo, Faci
 		return getResourcesManagerBl().createResource(sess, resource, vo, facility);
 	}
 
+	public Resource copyResource(PerunSession sess, Resource templateResource, Resource destinationResource, boolean withGroups) throws InternalErrorException, ResourceNotExistsException, FacilityNotExistsException, PrivilegeException, VoNotExistsException, ResourceExistsException {
+		Utils.checkPerunSession(sess);
+		Vo destinationVo = getPerunBl().getVosManagerBl().getVoById(sess, destinationResource.getVoId());
+		Facility destinationFacility = getPerunBl().getFacilitiesManagerBl().getFacilityById(sess, destinationResource.getFacilityId());
+		getPerunBl().getVosManagerBl().checkVoExists(sess, destinationVo);
+		getPerunBl().getFacilitiesManagerBl().checkFacilityExists(sess, destinationFacility);
+
+		Facility templateResourceFacility = getPerunBl().getFacilitiesManagerBl().getFacilityById(sess, templateResource.getFacilityId());
+
+		//check if user is part of template resource's facility
+		if(!AuthzResolver.isAuthorized(sess, Role.VOADMIN, templateResource) &&
+				!AuthzResolver.isAuthorized(sess,Role.FACILITYADMIN, templateResourceFacility)){
```

Comment:
VOADMIN is not allowed to see all settings of the Resource. For this reason, probably only FACILITYADMIN should be able to call this. VOADMIN is needed just for copying groups and group related attributes.